rhino/src/main/java/org/mozilla/javascript/Interpreter.java

```diff
@@ -296,6 +296,35 @@ void initializeArgs(
                 ScriptRuntime.initScript(fnOrScript, thisObj, cx, scope, desc.isEvalFunction());
             }
 
+            // For generators with default parameters, evaluate them after activation scope
+            // is created but BEFORE nested function declarations are added to the scope.
+            // This ensures that references to 'arguments' in default params get the arguments
+            // object, not a function declaration that might shadow it.
+            if (fnOrScript instanceof JSFunction) {
+                JSFunction jsFunc = (JSFunction) fnOrScript;
+                String[] generatorDefaultParams =
+                        jsFunc.getDescriptor().getGeneratorDefaultParams();
+                if (generatorDefaultParams != null && generatorDefaultParams.length > 0) {
+                    for (int i = 0; i < generatorDefaultParams.length; i += 2) {
```

Comment:
This seems to be a copy of wha we are doing in the opt runtime in this case. Maybe we have to have it only once